mono/metadata/threads-types.h

```diff
@@ -288,8 +288,7 @@ void mono_threads_perform_thread_dump (void);
 gboolean
 mono_thread_create_checked (MonoDomain *domain, gpointer func, gpointer arg, MonoError *error);
 
-/* Can't include utils/mono-threads.h because of the THREAD_INFO_TYPE wizardry */
-void mono_threads_add_joinable_runtime_thread (gpointer thread_info);
+void mono_threads_add_joinable_runtime_thread (MonoThreadInfo *thread_info);
```

Comment:
Still seems to me like this should be `struct _MonoThreadInfo`.  Otherwise aren't we depending on `mono/utils/mono-threads.h` to be included first?

Reply:
Yes. It is included first.